nio-impl/src/main/java/org/xnio/nio/NioTcpServer.java

```diff
@@ -188,7 +189,9 @@
                 handles[i].initializeTokenCount(i < tokens ? connections : 0);
             }
         }
-        mbeanHandle = NioXnio.register(new XnioServerMXBean() {
+        mbeanHandle = new XnioServerMXBean() {
+            private Closeable mbeanHandle = worker.registerServerMetrics(this);
```

Comment:
it is fully constructed as it doesn't hold any state as such, as that implementing methods do is operate on outer class itself.